File: repos_estudiantes_TARQUI_primera_entrega/ILIMORI_TOTOLA_ANEL_ESCARLET/PRACTICA 1/POLIMORFISMO/7/Parada.java
package siete;

public class Parada {
    private String[] admins = new String[10];
    private String[][] autos = new String[10][3]; 
    private String nombreP;

    private int nroAdmins = 0;
    private int nroAutos = 0;

    public Parada() {
        nombreP = "Sin nombre";
    }

    public Parada(String nombreP) {
        this.nombreP = nombreP;
    }

    public void adicionar(String admin) {
        if (nroAdmins < admins.length) {
            admins[nroAdmins] = admin;
            nroAdmins++;
        } else {
            System.out.println("No se pueden agregar más administradores.");
        }
    }

    public void adicionar(String modelo, String conductor, String placa) {
        if (nroAutos < autos.length) {
            autos[nroAutos][0] = modelo;
            autos[nroAutos][1] = conductor;
            autos[nroAutos][2] = placa;
            nroAutos++;
        } else {
            System.out.println("No se pueden agregar más autos.");
        }
    }

    public void mostrar() {
        System.out.println("===== PARADA: " + nombreP + " =====");
        
        System.out.println("\nAdministradores:");
        if (nroAdmins == 0) System.out.println("  (sin administradores)");
        for (int i = 0; i < nroAdmins; i++) {
            System.out.println("  - " + admins[i]);
        }

        System.out.println("\nAutos:");
        if (nroAutos == 0) System.out.println("  (sin autos)");
        for (int i = 0; i < nroAutos; i++) {
            System.out.println("  Modelo: " + autos[i][0] +
                               ", Conductor: " + autos[i][1] +
                               ", Placa: " + autos[i][2]);
        }
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/Polimorfismo/Ejercicio7/Parada.py
class Parada:
    def __init__(self, nombreP="Sin nombre", admins=None, autos=None):
        if admins is None:
            self.admins = [""] * 10
            self.nroAdmins = 0
        else:
            self.admins = admins
            self.nroAdmins = len(admins)

        if autos is None:
            self.autos = [["", "", ""] for _ in range(10)] 
            self.nroAutos = 0
        else:
            self.autos = autos
            self.nroAutos = len(autos)
        self.nombreP = nombreP
    def mostrar(self):
        print(f"=== Parada: {self.nombreP} ===")
        print("Administradores:")
        for i in range(self.nroAdmins):
            print(f"  - {self.admins[i]}")
        print("\nAutos:")
        for i in range(self.nroAutos):
            modelo, conductor, placa = self.autos[i]
            print(f"  Modelo: {modelo}, Conductor: {conductor}, Placa: {placa}")
    def adicionar(self, x):
        if self.nroAdmins < 10:
            self.admins[self.nroAdmins] = x
            self.nroAdmins += 1
            print(f"Administrador '{x}' agregado con éxito.")
        else:
            print("No se pueden agregar más administradores (límite 10).")
    def adicionar_auto(self, modelo, conductor, placa):
        if self.nroAutos < 10:
            self.autos[self.nroAutos] = [modelo, conductor, placa]
            self.nroAutos += 1
            print(f"Auto '{modelo}' agregado con éxito.")
        else:
            print("No se pueden agregar más autos (límite 10).")

if __name__ == "__main__":
    p1 = Parada()
    p1.mostrar()
    admins = ["Luis", "Marta"]
    autos = [["Toyota", "Juan", "ABC-123"], ["Nissan", "Ana", "XYZ-987"]]
    p2 = Parada("Terminal Norte", admins, autos)
    p2.mostrar()
    p2.adicionar("Carlos")
    p2.adicionar_auto("Kia", "Pedro", "JJJ-456")
    print("\nDespués de adicionar:")
    p2.mostrar()
